### ISSUE TITLE
Weird Looping Behavior on Windows

### ISSUE THREAD

**Issue by luvoid** [x-CK-x/Dataset-Curation-Tool#4]
### Describe the bug

Tested on both linux & windows. Only getting this issue on windows. It loops when calling functions from instantiated objects from other local files with the respective classes.

### Reproduction

Either downloading any data and/or trying to run inference with the captioning model creates this looping behavior with code in the webui.py script.

Tried upgrading it to gradio==3.32.0, but on windows10 the issue persists
the issue is OS related using gradio, but the loop behavior specifically happens when a function is called from a previously instantiated object in the webui.py.

* so with the batch_downloader class, running the download button
* as well as the the auto_tag class, running the interrogate button

**Comment by luvoid** [comment]
### Possible Cause Identified

Windows doesn't have a `fork` sys-command like Linux does. Python works around this by starting a new process, importing all scopes up to where the `mp.Process` was created, and continuing from there.

`webui.py` was using multiple processes, but the code that builds the Gradio UI is just sitting in `webui.py`, so everytime `webui.py` is imported, the UI hooks were created again.

tl;dr, every time mp.Process is called, `webui.py` is started from the beginning again in a new process.
